src/main/java/calculation/CalculateCoordinatesInterpolationFinal.java

```diff
@@ -0,0 +1,137 @@
+package calculation;
+
+import model.Location;
+import model.LocationPoint;
+import org.gavaghan.geodesy.Ellipsoid;
+import org.gavaghan.geodesy.GeodeticCalculator;
+import org.gavaghan.geodesy.GlobalCoordinates;
+import utils.Utilities;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class CalculateCoordinatesInterpolationFinal {
```

Comment:
You can change Final in the class name to something different :)
You can also make this method static.
I think most of the comments about moving variables outside to for loop would apply here as well

Or maybe you can just put all those algorithms in one class since they seems to share some variables/constants and logic. so each algo will be in separate static method.